packages/editor/src/components/post-title/post-title-rich.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * WordPress dependencies
+ */
+import { forwardRef, useState } from '@wordpress/element';
+import { ENTER } from '@wordpress/keycodes';
+import { useDispatch } from '@wordpress/data';
+import { pasteHandler } from '@wordpress/blocks';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import {
+	__unstableUseRichText as useRichText,
+	create,
+	toHTMLString,
+	insert,
+} from '@wordpress/rich-text';
+import { useMergeRefs } from '@wordpress/compose';
+import { __unstableStripHTML as stripHTML } from '@wordpress/dom';
+
+/**
+ * Internal dependencies
+ */
+import PostTypeSupportCheck from '../post-type-support-check';
+
+const PostTitleRich = forwardRef(
```

Comment:
This whole component could just be part of `index.js` if we'd prefer. It doesn't _really_ need to exist I suppose.